### ISSUE TITLE
FTransUNet：在计算混淆矩阵的时候为什么最后两列都是0，car类别的相关指标计算不出来

### ISSUE THREAD

**Issue by EvilGhostY** [sstary/SSRS#26]
你好，请问一下我在测试改代码的时候，发现计算的混淆矩阵后面两列为0，car有关的定量指标没有计算出来，请问是为什么呀？

Device : NVIDIA GeForce RTX 3090
load_pretrained: grid-size from 14 to 16
Load pretrained done.
203403558
training :  ['1', '3', '5', '7', '11', '13', '15', '17', '21', '23', '26', '28', '30', '32', '34', '37']
testing :  ['2', '4', '6', '8', '10', '12', '14', '16', '20', '22', '24', '27', '29', '31', '33', '35', '38']
BATCH_SIZE:  10
Stride Size:  256
E:\Anaconda3\envs\yolov5test\lib\site-packages\torch\nn\_reduction.py:42: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
E:\Anaconda3\envs\yolov5test\lib\site-packages\numpy\core\fromnumeric.py:3440: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
E:\Anaconda3\envs\yolov5test\lib\site-packages\numpy\core\_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
Train (epoch 1/50) [0/1000 (0%)]	Loss: 1.817661	Accuracy: 15.0909423828125
Train (epoch 1/50) [10/1000 (1%)]	Loss: 1.275442	Accuracy: 68.6981201171875
Train (epoch 1/50) [20/1000 (2%)]	Loss: 0.914344	Accuracy: 48.93951416015625
Train (epoch 1/50) [30/1000 (3%)]	Loss: 0.922132	Accuracy: 62.95013427734375
Train (epoch 1/50) [40/1000 (4%)]	Loss: 0.950104	Accuracy: 62.7471923828125
  0%|          | 0/17 [00:00<?, ?it/s]
  0%|          | 0/11 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/10 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/7 [00:00<?, ?it/s]
  0%|          | 0/8 [00:00<?, ?it/s]
  0%|          | 0/15 [00:00<?, ?it/s]
Confusion matrix :
[[16266563  1302474  4076816  1025815        0        0]
 [ 1615542 19014958   831858   487050        0        0]
 [ 3533812   862060  8636792  4338121        0        0]
 [  238238   310634  1984789 15978437        0        0]
 [  430874   169064   158750    36745        0        0]
 [  286882   194852   153546    57271        0        0]]
81991943 pixels processed
Total accuracy : 73.05
roads: 0.7175
buildings: 0.8663
low veg.: 0.4972
trees: 0.8631
cars: 0.0000
clutter: 0.0000
---
F1Score :
roads: 0.7223
buildings: 0.8682
low veg.: 0.5201
trees: 0.7903
cars: 0.0000
clutter: 0.0000
mean F1Score: 0.5802
---
Kappa: 0.6416
IoU :
roads: 0.5653
buildings: 0.7671
low veg.: 0.3514
trees: 0.6533
cars: 0.0000
clutter: 0.0000
mean IoU: 0.4674
---
[0.56526233 0.76708813 0.35142419 0.65332509 0.         0.        ]
mean MIoU old: 0.4674

**Comment by sstary** [comment]
car和clutter都是少样本类别，你这里可能是因为模型处于训练初期，可以多训练几个epoch观察一下。
如果一直都是0，可以更换一个其他模型，观察是普遍问题，还是FTransUNet的问题。如果是普遍问题，可以检查下标签数据加载是否正确。

**Comment by EvilGhostY** [comment]
好的，感谢。我在观察一下。

**Comment by EvilGhostY** [comment]
后面几个epoch就出现了，感谢您的回复。目前正在做相关的研究。

**Comment by sstary** [comment]
好的，祝顺利

**Comment by EvilGhostY** [comment]
你好，还有一个问题对于POTSDAM和Vaihingen数据集，他们训练的epoch都是50吗？我更改了数据集的分配跟原论文不一样，是否需要进行更改？

**Comment by sstary** [comment]
是的，我们对所有的模型使用了一致的超参数下。50epoch实际上训练了很多轮，你这种数据集划分也可以使用这组超参数。

**Comment by sstary** [comment]
如果对你有帮助，请点个star。
